src/lib.rs

```diff
@@ -319,6 +341,88 @@ pub trait MapInsert<K>: Collection {
 	fn insert(&mut self, key: K, value: Self::Item) -> Self::Output;
 }
 
+/// Mutable map that supports the entry api
+pub trait EntryApi: Keyed {
```

Comment:
I would prefer sticking with the name of the trait matching the name of the function (`Entry`).

Reply:
I chose the name EntryApi to avoid conflicting with the name of the returned enum Entry.  I could possibly give them the same name if I keep the trait in the root, but this would make it awkward to use the trait and enum at the same time.

Reply:
Right, I didn't consider that. Maybe `GetEntry`? But I guess `EntryApi` is also fine if you want to keep it.